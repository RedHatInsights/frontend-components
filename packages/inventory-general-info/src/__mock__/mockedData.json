{
    "total": 1,
    "count": 1,
    "page": 1,
    "per_page": 50,
    "results": [
        {
            "id": "6de30d78-f052-41d4-b123-125017443a98",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "7.4",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu"
            }
        }
    ]
}
