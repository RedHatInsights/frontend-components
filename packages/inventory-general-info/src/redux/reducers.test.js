import toJson from 'enzyme-to-json';
import {
    formatBytes,
    sizes,
    calculateRepos,
    calculateInterfaces,
    systemProfilePending,
    onSystemProfile
} from './reducers';
import mockedData from '../__mock__/mockedData.json';

describe('formatBytes', () => {
    it('should return 0 B for empty values', () => {
        expect(formatBytes('a')).toBe('0 B');
        expect(formatBytes(0)).toBe('0 B');
        expect(formatBytes()).toBe('0 B');
    });

    sizes.map((oneSize, key) => {
        it(`size ${oneSize}`, () => {
            const formatted = formatBytes(Number(511 + '0'.repeat(3 * key)));
            expect(formatted.substring(formatted.length - 2)).toBe(`${key === 0 ? ' ' : ''}${oneSize}`);
        });
    });
});

it('should calculate repos', () => {
    const repos = calculateRepos([{}, {
        enabled: true
    }, {
        enabled: false
    }]);
    expect(repos.disabled.length).toBe(2);
    expect(repos.enabled.length).toBe(1);
});

it('should calculate NIC', () => {
    const interfaces = calculateInterfaces([{
        ipv4_addresses: [ '1' ] // eslint-disable-line camelcase
    }, {
        ipv4_addresses: [ '1' ], // eslint-disable-line camelcase
        ipv6_addresses: [ '1' ] // eslint-disable-line camelcase
    }, {
        ipv6_addresses: [ '1' ] // eslint-disable-line camelcase
    }]);
    expect(interfaces.interfaces.length).toBe(3);
    expect(interfaces.ipv4.length).toBe(2);
    expect(interfaces.ipv6.length).toBe(2);
});

describe('systemProfilePending', () => {
    it('no state', () => {
        expect(systemProfilePending()).toEqual({
            systemProfile: {
                loaded: false
            }
        });
    });

    it('should\'t rewrite state', () => {
        expect(systemProfilePending({
            test: 'data'
        })).toEqual({
            test: 'data',
            systemProfile: {
                loaded: false
            }
        });
    });
});

describe('onSystemProfile', () => {
    it('empty results', () => {
        expect(onSystemProfile(undefined, { payload: {} })).toEqual({
            systemProfile: {
                loaded: true,
                network: undefined,
                ramSize: '0 B',
                repositories: undefined
            } });
    });
    describe('correct data', () => {
        const { systemProfile: { network, repositories, ramSize, ...rest } } = onSystemProfile(undefined, { payload: mockedData });

        it('network', () => {
            expect(network.interfaces.length).toBe(1);
            expect(network.ipv4.length).toBe(1);
            expect(network.ipv6.length).toBe(1);
        });

        it('repositories', () => {
            expect(repositories.disabled.length).toBe(1);
            expect(repositories.enabled.length).toBe(0);
        });

        it('ramSize', () => {
            expect(ramSize).toBe('488.35 MB');
        });

        it('rest data should match', () => {
            expect(rest).toMatchSnapshot();
        });
    });
});
