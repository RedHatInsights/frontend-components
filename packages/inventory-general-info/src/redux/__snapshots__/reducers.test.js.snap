// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onSystemProfile correct data rest data should match 1`] = `
Object {
  "arch": "x86_64",
  "bios_release_date": "04/01/2014",
  "bios_vendor": "SeaBIOS",
  "bios_version": "1.10.2-2.fc27",
  "cores_per_socket": 1,
  "cpu_flags": Array [
    "fpu",
  ],
  "enabled_services": Array [
    "auditd",
  ],
  "infrastructure_type": "virtual",
  "infrastructure_vendor": "qemu",
  "installed_packages": Array [
    "0:GeoIP-1.5.0-11.el7",
  ],
  "installed_services": Array [
    "arp-ethers",
  ],
  "kernel_modules": Array [
    "veth",
  ],
  "last_boot_time": "2018-03-14T17:06:10",
  "loaded": true,
  "network_interfaces": Array [
    Object {
      "ipv4_addresses": Array [
        "127.0.0.1",
      ],
      "ipv6_addresses": Array [
        "::1",
      ],
      "mac_address": "00:00:00:00:00:00",
      "mtu": 65536,
      "name": "lo",
      "state": "UNKNOWN",
      "type": "loopback",
    },
  ],
  "number_of_cpus": 1,
  "number_of_sockets": 1,
  "os_kernel_version": "3.10.0",
  "os_release": "7.4",
  "running_processes": Array [
    "qmgr",
  ],
  "satellite_managed": false,
  "system_memory_bytes": 512073728,
  "yum_repos": Array [
    Object {
      "enabled": false,
      "gpgcheck": true,
      "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
    },
  ],
}
`;
